File: src/C#/CreationalPatterns/Singleton/VotingMachine/VotingMachineV3.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace VotingMachine
{
    public sealed class VotingMachineV3
    {
        private int totalVotes = 0;

        private static readonly VotingMachineV3 votingMachineInstance = new VotingMachineV3();

        private VotingMachineV3()
        {

        }

        public int TotalVotes => totalVotes;

        public static VotingMachineV3 VotingMachineInstance => votingMachineInstance;

        public void RegisterVote()
        {
            totalVotes++;
            Console.WriteLine($"Registered Vote #{totalVotes}");
        }
    }
}

File: src/Java/CreationalPatterns/Singleton/VotingMachine/src/VotingMachineV3.java
public final class VotingMachineV3 {
    private int totalVotes = 0;

    private static final VotingMachineV3 votingMachineInstance = new VotingMachineV3();

    private VotingMachineV3() {

    }

    public int getTotalVotes() {
        return totalVotes;
    }

    public static VotingMachineV3 getVotingMachineInstance() {
        return votingMachineInstance;
    }

    public void registerVote() {
        totalVotes++;
        System.out.println("Registered Vote #" + totalVotes);
    }
}
